idaes/core/surrogate/pysmo/tests/test_sampling.py

```diff
@@ -2240,5 +2241,631 @@ def test_sample_points_02(self, array_type):
             )
 
 
+class TestCustomSampling:
+    input_array = [[x, x + 10, (x + 1) ** 2 + x + 10] for x in range(10)]
+    input_array_list = [[x, x + 10, (x + 1) ** 2 + x + 10] for x in range(2)]
+    y = np.array(
+        [
+            [i, j, ((i + 1) ** 2) + ((j + 1) ** 2)]
+            for i in np.linspace(0, 10, 21)
+            for j in np.linspace(0, 10, 21)
+        ]
+    )
+    full_data = {"x1": y[:, 0], "x2": y[:, 1], "y": y[:, 2]}
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("array_type", [np.array, pd.DataFrame])
+    def test__init__selection_01(self, array_type):
```

Comment:
Could you use more meaningful test names here? E.g. `test_selection_no_samples` instead of `test__init__selection_03`.

This makes it much easier to debug in future if something goes wrong as the name of the test tells you something about what should have happened.

Reply:
Done